### PR TITLE
feat(input-number): add suffix prop & change controls animation

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
resolve https://github.com/ant-design/ant-design/issues/49621

### 💡 Background and solution
chrome-capture-2024-7-2 (1)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  InputNumber supports `suffix` prop now.       |
| 🇨🇳 Chinese |  InputNumber 支持 `suffix` 属性。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input-number/style/index.ts [hunk 1]
```diff
@@ -452,15 +456,22 @@ const genAffixWrapperStyles: GenerateStyle<InputNumberToken> = (token: InputNumb
...
+          transition: `margin ${motionDurationMid}`,
         },
       },
+
+      [`&:hover ${componentCls}-handler-wrap, &-focused ${componentCls}-handler-wrap`]: {
+        width: token.handleWidth,
+        opacity: 1,
+      },
+      [`&:hover ${componentCls}-suffix`]: {
+        marginInlineEnd: `calc(${token.handleWidth} + ${paddingInline})`,
```
Comment (li-jia-nan) [inline components/input-number/style/index.ts:473]: 这里不能只能用 calc 计算吧，参考其它地方

Comment (coding-ice) [reply components/input-number/style/index.ts:473]: 请教一下，是这样不
`token.calc(token.handleWidth).add(paddingInline).equal()`

**Comment by afc163** [comment]
覆盖率掉了。

File: components/input-number/style/index.ts (hunk 1 above)
Comment (li-jia-nan) [reply components/input-number/style/index.ts:473]: 看起来没啥问题，要看看 F12 是否正确

**Comment by afc163** [comment]
ci 挂了

**Review by afc163 — APPROVED** [review]

**Pull request merged**